casa.py

```diff
@@ -25,8 +26,13 @@ def _get(self, q, params=''):
 
         if (q[-1] == '/'): q = q[:-1]
 
-        r = requests.get('{url}{q}?{params}'.format(url=self.url, q=q, params=params),
-                headers=self.headers)
+        retries = 0
+
+        while (retries < self.api_max_retries):
+            r = requests.get('{url}{q}?{params}'.format(url=self.url, q=q, params=params),
```

Comment:
Admittedly, I'm not a python exception flow master, but does this need to be wrapped in an equivalent try/catch block?

Traceback (most recent call last):
  File "./assigner.py", line 220, in <module>
    main()
  File "./assigner.py", line 66, in main
    autocasa(bapi, capi, bcfg, ccfg, args.dry_run)
  File "./assigner.py", line 93, in autocasa
    comments = bapi.get_comments(bug.get('id'))['bugs'][str(bug.get('id'))]['comments']
  File "/usr/lib/python3.6/site-packages/bugzilla.py", line 55, in get_comments
    return self._get('bug/{bugid}/comment'.format(bugid=bugid))
  File "/usr/lib/python3.6/site-packages/bugzilla.py", line 123, in _get
    raise Exception(r.url, r.reason, r.status_code, r.json())
Exception: ('https://bugzilla.mozilla.org/rest/bug/1456277/comment?api_key=****', 'OK', 200, {'documentation': 'https://bmo.readthedocs.org/en/latest/api/', 'error': True, 'code': 100500, 'message': "\nCan't connect to the database.\nError: Lost connection to MySQL server at 'reading initial communication packet', system error: 104\n  Is your database installed and up and running?\n  Do you have the correct username and password selected in localconfig?\n\n"})
Build step 'Execute shell' marked build as failure

Reply:
this is raising my own exception in this example, not a request exception
that means all retries have been done or thats something you ran before?